backend/src/routing/controllers/api/Abitur.ts

```diff
@@ -11,9 +12,44 @@ export default class Abitur {
     static POSTapplyForTopic(req: Request, res: Response): void {
         if (rejectWhenValidationsFail(req, res)) return;
 
-        const {responsibleTeacherId, art, bezugsfachId, partnerStudentId, referenzfachId, topic} = req.body;
-        const sql = 'INSERT INTO abiturpruefungen (betreuendeLehrkraftID, art, bezugsfachID, partnerStudentId, referenzfachID, thema) VALUES (?,?,?,?,?,?)';
-        insertData(sql, [responsibleTeacherId, art, bezugsfachId, partnerStudentId, referenzfachId, topic], defaultInsertCallback(res));
+        let sql = 'SELECT id FROM abiturpruefungen WHERE studentID = ?';
+        getAllResults(sql, [getStudentId()], (a, err) => {
+            if (err) {
+                res.status(500).json(err.name);
+            } else {
+                if (a.length > 0) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    updateUpdateColumns(a[0].id);
+                } else {
+                    insertFirstSubmission();
+                }
+            }
+        });
+
+        const updateUpdateColumns = (id: number) => {
+            const {updatedExaminer, updatedBezugsfach, updatedPartnerStudentName, updatedReferenzfach, updatedTopicArea, problemQuestion, updatedProblemQuestion, presentationForm, updatedPresentationForm} = req.body;
+            let setString = '';
+            const args = [];
+            setString += ' updatedExaminer = ?'; args.push(updatedExaminer ? updatedExaminer : null);
+            setString += ', updatedBezugsfach = ?'; args.push(updatedBezugsfach ? updatedBezugsfach : null);
```

Comment:
Das könnte man sicher gut auslagern, ist gerade ein bisschen WET. Aber besteht erstmal kein Änderungsbedarf.

Reply:
Was heißt WET?

Reply:
WET - We enjoy typing
DRY - Don't repeat yourself

Du bist hier doch der clean code experte, als ob du noch nie davon gehört hast. :D